src/StrokeEngine.cpp

```diff
@@ -557,169 +183,43 @@ String StrokeEngine::getPatternName(int index) {
-
-                // drive free of switch and set axis to lower end
-                servo->moveTo(_minStep);
-
-            } else {
-                servo->forceStopAndNewPosition(_motor->stepsPerMillimeter * (_physics->physicalTravel - _physics->keepoutBoundary));
-
-                // drive free of switch and set axis to front end
-                servo->moveTo(_maxStep);
-            }
-            _isHomed = true;
-
-            // drive free of switch and set axis to 0
-            servo->moveTo(0);
-            
-            // Break loop, home was found
-            break;
-        }
-
-        // Pause the task for 20ms to allow other tasks
-        vTaskDelay(20 / portTICK_PERIOD_MS);
-    }
-    
-    // disable Servo if homing has not found the homing switch
-    if (!_isHomed) {
-        servo->disableOutputs();
-        _state = UNDEFINED;
-
-#ifdef DEBUG_TALKATIVE
-        Serial.println("Homing failed");
-#endif
-
-    } else {
-        // Set state to ready
-        _state = READY;
-
-#ifdef DEBUG_TALKATIVE
-        Serial.println("Homing succeeded");
-#endif
-    }
-
-    // Call notification callback, if it was defined.
-    if (_callBackHomeing != NULL) {
-        _callBackHomeing(_isHomed);
-    }
-
-    // Set first point for telemetry
-    if (_callbackTelemetry != NULL) {
-        _callbackTelemetry(0.0, 0.0, false);
-    }
-
-#ifdef DEBUG_TALKATIVE
-    Serial.println("Stroke Engine State: " + verboseState[_state]);
-#endif
-
-    // delete one-time task
-    _taskHomingHandle = NULL;
-    vTaskDelete(NULL);
-}
-
 void StrokeEngine::_stroking() {
     motionParameter currentMotion;
 
     while(1) { // infinite loop
 
-        // Suspend task, if not in PATTERN state
-        if (_state != PATTERN) {
-            vTaskSuspend(_taskStrokingHandle);
+        // Suspend task, if motor is not active
+        // TODO - This doesn't look quite right
+        if (!this->motor->isInState(MotorState::ACTIVE)) {
+          ESP_LOGI("StrokeEngine", "Motor is no longer active! Attempting to suspend pattern.");
+          this->stopPattern();
         }
 
         // Take mutex to ensure no interference / race condition with communication threat on other core
         if (xSemaphoreTake(_patternMutex, 0) == pdTRUE) {
 
-            if (_applyUpdate == true) {
+            if (this->applyUpdate == true) {
                 // Ask pattern for update on motion parameters
                 currentMotion = patternTable[_patternIndex]->nextTarget(_index);
-            
-                // Increase deceleration if required to avoid crash
-                if (servo->getAcceleration() > currentMotion.acceleration) {
-#ifdef DEBUG_CLIPPING
-                    Serial.print("Crash avoidance! Set Acceleration from " + String(currentMotion.acceleration));
-                    Serial.println(" to " + String(servo->getAcceleration()));
-#endif
-                    currentMotion.acceleration = servo->getAcceleration();
-                }

```

Comment:
Yeah, this would be handled by the Motors themselves. I think originally I thought the LinMots wouldn't suffer from this due to it's internal motion planner, but after thinking on it a bit more, I'm not so sure. It seems like a generic problem where if the deceleration is slow enough and the current speed fast enough, you won't be able to slow to a stop. LinMot might some some protection against this, but I'm not sure?

I'll probably build some generic handlers in the `goToPos` function to help prevent these scenarios.

Thanks for calling this out, as I think I had made a bad assumption with the LinMots

Reply:
But even if the linmot has, generic steppers and the JMC servo won't. Making this generic seems a sound decision.